ctapipe/io/eventsource.py

```diff
@@ -298,8 +295,18 @@ def _find_compatible_source(cls, input_url):
         available_classes = non_abstract_children(cls)
 
         for subcls in available_classes:
-            if subcls.is_compatible(input_url):
-                return subcls
+            try:
+                if subcls.is_compatible(input_url):
+                    return subcls
+            except Exception as e:
+                warnings.warn(f"{subcls.__name__}.is_compatible raised exception: {e}")
+
+        # provide a more helpful error for non-existing input_url
+        if not input_url.exists():
+            raise TraitError(
+                f"input_url {input_url} is not an existing file "
+                " and no EventSource implementation claimed compatibility"
```

Comment:
I think this error would benefit from naming the available eventsources like the error below

Reply:
I explicitly left it out because it is right now kind of assumed that for a non-existing file, the error is a typo / wrong path and people should see that immediately and not be confused by a long list of tried sources.

But I don't have a strong opinion here.

Reply:
I agree